IndividualMiniprojectC++/src/RouteController.cpp

```diff
@@ -46,13 +48,47 @@ void RouteController::retrieveAppointment(const crow::request& req, crow::respon
     }
 }
 
+void RouteController::createAppointment(const crow::request& req, crow::response& res) {
+    try {
+        auto title = req.url_params.get("title");
+        auto startTime = std::stoi(req.url_params.get("startTime"));
+        auto endTime = std::stoi(req.url_params.get("endTime"));
+        auto location = req.url_params.get("location");
+        std::string apptCode = "APPT" + std::to_string(apptCodeCounter++);
+
+        auto appointmentMapping = myFileDatabase->getAppointmentMapping();
+
+        auto it = appointmentMapping.find(apptCode);
+        if (it == appointmentMapping.end()) {
+            res.code = 200;
+            res.write("Appointment Created : apptCode ");
+            res.write(apptCode);
+            Appointment appt1(apptCode, title, startTime, endTime, location);
+            std::map<std::string, Appointment> apptMapping;
+            apptMapping[apptCode] = appt1;
+            myFileDatabase->setApptMapping(apptMapping);
```

Comment:
Wouldn't this cause all existing appointments to be lost?

Reply:
Yes, the appointmentMapping from myFileDatabase has to be used. Fixed.